src/sshfs_mount/sshfs_mount.cpp

```diff
@@ -122,26 +124,31 @@ auto get_sshfs_exec_and_options(mp::SSHSession& session)
 // Split a path into existing and to-be-created parts.
 std::pair<std::string, std::string> get_path_split(mp::SSHSession& session, const std::string& target)
 {
-    QDir complete_path(QString::fromStdString(target));
-    QString absolute;
+    std::string absolute;
 
-    if (complete_path.isRelative())
+    switch (target[0])
     {
-        std::string home = run_cmd(session, "pwd");
-        mp::utils::trim_newline(home);
-        absolute = QString::fromStdString(home) + '/' + complete_path.path();
-    }
-    else
-    {
-        absolute = complete_path.path();
+    case '~':
+        absolute =
+            run_cmd(session, fmt::format("echo ~{}", mp::utils::escape_for_shell(target.substr(1, target.size() - 1))));
+        mp::utils::trim_newline(absolute);
+        break;
+    case '/':
+        absolute = target;
+        break;
+    default:
+        absolute = run_cmd(session, fmt::format("echo $PWD/{}", mp::utils::escape_for_shell(target)));
+        mp::utils::trim_newline(absolute);
+        break;
     }
 
-    std::string existing = run_cmd(
-        session,
-        fmt::format("sudo /bin/bash -c 'P=\"{}\"; while [ ! -d \"$P/\" ]; do P=${{P%/*}}; done; echo $P/'", absolute));
+    std::string existing =
+        run_cmd(session, fmt::format("sudo /bin/bash -c 'P={}; while [ ! -d $P/ ]; do P=${{P%/*}}; done; echo $P/'",
+                                     mp::utils::escape_for_shell(absolute)));
```

Comment:
We do need the quotes around all the instances of `$P`, as it will be unescaped.

Reply:
Oh, right... there is no ~ there, so we can safely return to using quotes.

I also added a test for escaping.